src/crypto/slow-hash.c

```diff
@@ -908,7 +908,13 @@ STATIC INLINE void aes_pseudo_round_xor(const uint8_t *in, uint8_t *out, const u
 void cn_slow_hash(const void *data, size_t length, char *hash, int variant, int prehashed)
 {
     RDATA_ALIGN16 uint8_t expandedKey[240];
+
+#ifndef FORCE_USE_HEAP
     RDATA_ALIGN16 uint8_t hp_state[MEMORY];
+#else
+    RDATA_ALIGN16 uint8_t *hp_state = NULL;
+    hp_state = (uint8_t *)malloc(MEMORY);
```

Comment:
Why the double assignment ?

Reply:
i am no friend of it - just followed the pattern already used in the file:
https://github.com/monero-project/monero/blob/8361d60aef6e17908658128284899e3a11d808d4/src/crypto/slow-hash.c#L1130-L1131
can change mine or all - as you say.

Reply:
I don't recall mind tbh. It just feels redundant.